### PR TITLE
Dont assume json payloads, provide full record details

### PR DESCRIPTION
**Pull request by msurdi**
this PR fixes https://github.com/ticketea/pynesis/issues/4 and https://github.com/ticketea/pynesis/issues/3

### CODE REVIEW

File: pynesis/streams.py [hunk 1]
```diff
@@ -59,27 +49,29 @@ def id(self):  # type ()->str
...
+        self.sequence_number = raw_record.get("SequenceNumber")  # type: str
+        self.approximate_arrival_timestamp = raw_record.get("ApproximateArrivalTimestamp")  # type: datetime
+        self.data = raw_record.get("Data")  # type: bytes
+        self.partition_key = raw_record.get("PartitionKey")  # type: str
 
-    @property
-    def sequence(self):  # type ()-> str
-        return str(self._raw_record.get("SequenceNumber", ""))
+    @staticmethod
+    def build(sequence_number, approximate_arrival_timestamp, data,
```
Comment (Kartones) [inline pynesis/streams.py]: shouldn't it be an integer for sequence_number?

File: pynesis/streams.py [hunk 2]
```diff
@@ -225,25 +217,29 @@ class DummyStream(Stream):
...
 
-    def read(self):  # type: ()->Generator[Dict, None, None]
+    def read(self):  # type: ()->Generator[KinesisRecord, None, None]
         fake_values = self._fake_values  # type: Iterable
         if self._loop:
             fake_values = cycle(self._fake_values)
 
-        for message in fake_values:
-            yield message
+        for i, message in enumerate(fake_values):
```
Comment (patoroco) [inline pynesis/streams.py]: `index` would be a better name for this variable IMHO

File: pynesis/streams.py (hunk 1 above)
Comment (msurdi) [inline pynesis/streams.py]: Nope! It's a trap! boto documentation says it is a string, despite the word "number" in the name. And the api is actually returning an `str`. See [here](http://boto3.readthedocs.io/en/latest/reference/services/kinesis.html#Kinesis.Client.get_records)

File: pynesis/streams.py (hunk 2 above)
Comment (msurdi) [inline pynesis/streams.py]: I accept boats (aceptamos barco)

File: pynesis/streams.py (hunk 1 above)
Comment (Kartones) [inline pynesis/streams.py]: :sob: Ok then

**Pull request merged**